core/src/test/java/org/apache/iceberg/TestClientPoolImpl.java

```diff
@@ -84,6 +141,18 @@ static class RetryableException extends RuntimeException {}
 
   static class NonRetryableException extends RuntimeException {}
 
+  static class ImplementationSpecificException extends NonRetryableException {
```

Comment:
```suggestion
  static class CustomException extends NonRetryableException {
```
this reads shorter and also reduces test method name lengths

Reply:
done